corehq/apps/geospatial/management/commands/index_geolocation_case_properties.py

```diff
@@ -0,0 +1,86 @@
+import math
+
+from django.core.management.base import BaseCommand
+
+from dimagi.utils.chunked import chunked
+
+from corehq.apps.es import CaseSearchES, case_search_adapter, filters, queries
+from corehq.apps.es.case_search import (
+    CASE_PROPERTIES_PATH,
+    PROPERTY_GEOPOINT_VALUE,
+    PROPERTY_KEY,
+)
+from corehq.apps.es.client import manager
+from corehq.apps.geospatial.utils import get_geo_case_property
+from corehq.form_processor.models import CommCareCase
+from corehq.util.log import with_progress_bar
+
+DEFAULT_QUERY_LIMIT = 10_000
+DEFAULT_CHUNK_SIZE = 100
+
+
+class Command(BaseCommand):
+    help = 'Index geopoint data in ES'
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('--case_type', required=False)
+        parser.add_argument('--query_limit', required=False)
+        parser.add_argument('--chunk_size', required=False)
+
+    def handle(self, *args, **options):
+        domain = options['domain']
+        case_type = options.get('case_type')
+        query_limit = options.get('query_limit')
+        chunk_size = options.get('chunk_size')
+        index_case_docs(domain, query_limit, chunk_size, case_type)
+
+
+def index_case_docs(domain, query_limit=DEFAULT_QUERY_LIMIT, chunk_size=DEFAULT_CHUNK_SIZE, case_type=None):
+    geo_case_property = get_geo_case_property(domain)
+    query = _es_case_query(domain, geo_case_property, case_type)
+    count = query.count()
+    print(f'{count} case(s) to process')
+    batch_count = math.ceil(count / query_limit)
```

Comment:
We should probably just do some sanity checking so that's it's not zero, since it would raise an error here. It would be OK as is if it's ever only going to be a developer running this command, but since I remember a ticket coming that we're going to invoke this programmatically I wonder if it's not best to do these types of checks as well.

Reply:
That's a good point. In the follow-up work we'll still have control over where this gets executed and what params to give it, however I do agree it's good to do some sanity checking here since this could be used outside the scope of our follow-up work.

@Charl1996 A question I have however, is should we handle this gracefully, or return with an assert error that needs to be fixed? My thinking is the former, since I assume if a dev passed in a param of 0 they mean to say don't add a limit so we can simply make the `batch_count` 1 and pass `None` in for `size` when fetching the query.

Reply:
I think we can probably work with those assumptions.

Reply:
Cool, I have addressed this in [f9f78b1](https://github.com/dimagi/commcare-hq/pull/35090/commits/f9f78b1c4da1778c215c431ae32a31bcddef8194).